functorch/csrc/BatchRulesReduceOps.cpp

```diff
@@ -275,6 +275,101 @@ std::tuple<Tensor, optional<int64_t>, Tensor, optional<int64_t>> aminmax_batchin
   return std::make_tuple(min, 0, max, 0);
 }
 
+std::tuple<Tensor, Tensor> cdist_batch_rule_helper(
+    const Tensor& x1, optional<int64_t> x1_bdim,
+    const Tensor& x2, optional<int64_t> x2_bdim) {
+
+  auto x1_ = moveBatchDimToFront(x1, x1_bdim);
+  auto x2_ = moveBatchDimToFront(x2, x2_bdim);
+
+  int64_t d1 = static_cast<int64_t>(x1_bdim.has_value());
+  int64_t d2 = static_cast<int64_t>(x2_bdim.has_value());
+  while (x2_bdim && x1_.dim() - d1 > x2_.dim() - d2) {
+    // x1:                      [  S1, S2, .............., SK]
+    // x2: [B, D1, ..., DN]  -> [B, 1,  1, ..., 1, D1 ..., DN]
+    x2_ = x2_.unsqueeze(1);
+  }
+  while (x1_bdim && x2_.dim() - d2 > x1_.dim() - d1) {
+    // x2:                      [  S1, S2, .............., SK]
+    // x1: [B, D1, ..., DN]  -> [B, 1,  1, ..., 1, D1 ..., DN]
+    x1_ = x1_.unsqueeze(1);
+  }
+  // Backward requires contiguous tensors
+  if (!x1_.is_contiguous()) {
+    x1_ = x1_.contiguous();
+  }
+  if (!x2_.is_contiguous()) {
+    x2_ = x2_.contiguous();
+  }
+  return std::make_tuple(x1_, x2_);
+}
+
+std::tuple<Tensor,optional<int64_t>> cdist_batch_rule(
+    const Tensor& x1, optional<int64_t> x1_bdim,
+    const Tensor& x2, optional<int64_t> x2_bdim,
+    const double p, c10::optional<int64_t> compute_mode) {
+
+  auto x12 = cdist_batch_rule_helper(x1, x1_bdim, x2, x2_bdim);
+  return std::make_tuple(at::cdist(std::get<0>(x12), std::get<1>(x12), p, compute_mode), 0);
+}
+
+std::tuple<Tensor,optional<int64_t>> cdist_forward_batch_rule(
+    const Tensor& x1, optional<int64_t> x1_bdim,
+    const Tensor& x2, optional<int64_t> x2_bdim,
+    const double p, c10::optional<int64_t> compute_mode) {
+
+  auto x12 = cdist_batch_rule_helper(x1, x1_bdim, x2, x2_bdim);
```

Comment:
cdist_batch_rule_helper returns tensors that are more than 3 dimensional. Then we call `_cdist_forward` in the following line.

In the [PyTorch documentation for cdist](https://pytorch.org/docs/stable/generated/torch.cdist.html) it seems to imply that cdist accepts tensors with 3 dimensions only. Is the PyTorch documentation wrong?

Reply:
Reading the OpInfo for this, it does look like the pytorch documentation is wrong unless I am reading the incorrect page

Reply:
Yes, the docs is incomplete about what could be accepted as input. I think it should be something like
- x1: *, P, M
- x2: *, R, M